core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/DataSource.scala

```diff
@@ -77,33 +75,15 @@ class SlickDataSource(val databaseConfig: DatabaseConfig[JdbcProfile])(implicit
     }
 
     database.run(callerActionWithTempTables.withPinnedSession).recover { case t: Throwable =>
-      logTransactionError(t, tempTableTypes)
+      // log the transaction error at DEBUG level. Since we rethrow the error, calling code is responsible for handling
+      // it and, if appropriate, logging at ERROR level. Only ERROR-level logs are picked up as Sentry alerts.
+      logger.debug(
+        s"Transaction with temporary tables failed for (${tempTableTypes.mkString(",")}). ${t.getClass.getName}: ${t.getMessage}"
+      )
       throw t
```

Comment:
Did you consider throwing a new exception that has the message with `tempTableTypes` in it? Or is the worry there that we don't know if some caller is depending on the current exception being thrown, and changing that could break that code?

Reply:
> we don't know if some caller is depending on the current exception being thrown, and changing that could break that code?

exactly this!